extensions/amp-form/0.1/amp-form.js

```diff
@@ -146,6 +147,16 @@ export class AmpForm {
     /** @private {?string} */
     this.state_ = null;
 
+    // Update the form non-xhr action to add `__amp_source_origin` parameter.
+    // This allows publishers to understand where the request is coming from.
+    const action = this.form_.getAttribute('action');
+    user().assert(action, 'form action is required: %s',
+        this.form_);
+    assertHttpsUrl(action, this.form_, 'action');
+    user().assert(!startsWith(action, urls.cdn),
+        'form action should not be on cdn.ampproject.org: %s', this.form_);
+    this.form_.setAttribute('action', getCorsUrl(this.win_, action));
```

Comment:
hmm. I can add this in `document-submit.js` instead if we think this is important.